src/encounter_api.c

```diff
@@ -58,9 +59,89 @@ ApiStatus MakeNpcs(Evt* script, s32 isInitialCall) {
     return ApiStatus_BLOCK;
 }
 
-INCLUDE_ASM(s32, "encounter_api", RemoveNpc, Evt* script, s32 isInitialCall);
+ApiStatus RemoveNpc(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    Enemy* enemy = script->owner1.enemy;
+    ApiStatus ret = ApiStatus_DONE1;
+    s32 id = evt_get_variable(script, *args++);
+    EncounterStatus* currentEncounter = &gCurrentEncounter;
+    s32 i, j;
+
+    if ((s32)enemy != NPC_SELF) {
```

Comment:
wtf

Reply:
To properly do this, we'd want to name and typedef the owner1 / owner2 types so we can refer to these unions themselves instead of casting stuff. but I think it's outside the scope of this